### PR TITLE
Added Swagger Documentation and API Versioning

### PR DESCRIPTION
**Pull request by LBHDTortora**
Merge all conflicts between master and swagger branch.
Kept logic and new implementation from master and added implementation of Swagger Doc Gen and API Versioning.

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{
```
Comment (LBHmsaitz) [inline .vscode/launch.json]: This can be removed from the project

File: .vscode/tasks.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (LBHmsaitz) [inline .vscode/tasks.json]: This can be removed as well, it also could be added to the gitignore file

File: property-api/Startup.cs [hunk 3]
```diff
@@ -22,11 +32,75 @@ public Startup(IConfiguration configuration)
...
+        private static List<ApiVersionDescription> _apiVersions { get; set; }
 
         // This method gets called by the runtime. Use this method to add services to the container.
         public void ConfigureServices(IServiceCollection services)
         {
             services.AddMvc().SetCompatibilityVersion(CompatibilityVersion.Version_2_1);
+
+            services.AddSingleton<IApiVersionDescriptionProvider, DefaultApiVersionDescriptionProvider>();
+
+            services.AddApiVersioning(o =>
```
Comment (LBHmsaitz) [inline property-api/Startup.cs]: Suggest moving the swagger config to private methods to make the ConfigureServices method easier to read

File: .vscode/launch.json (hunk 1 above)
Comment (LBHDTortora) [reply .vscode/launch.json]: Removed from the PR, will add on .gitignore

File: property-api/Startup.cs (hunk 3 above)
Comment (LBHDTortora) [reply property-api/Startup.cs]: Removed from the PR, will add on .gitignore

File: .vscode/tasks.json (hunk 2 above)
Comment (LBHDTortora) [reply .vscode/tasks.json]: Removed from the PR, will add on .gitignore

File: property-api/Startup.cs (hunk 3 above)
Comment (LBHDTortora) [reply property-api/Startup.cs]: Method has been extracted. There was a conflict with the HttpGet method in Swagger with an explicit route triggering an exception. I have removed the route from the HttpGet attribute and kept the route on the [Route] attribute. This seems to have fixed the problem.

**Comment by LBHDTortora** [comment]
Implemented Swagger Doc and Gen to Property API and included xml documentation to be generated on controllers methods.
Fixed a swagger conflicts with Http method where the [HttpGet] attribute could not include routing params (ex. [HttpGet('{params}')].
Params are still declared in the Route attribute. [Route('{params}')].

**Pull request merged**